### PR TITLE
MCD: Add support for RHCOS9 and SCOS9

### PR DESCRIPTION
**Pull request by LorbusChris**
**- What I did**

This adds MCD support for `ID="scos"` in `/usr/lib/os-release` and
`/etc/os-release`.

Done:
- [x] discuss removal of kernel-arg allowlist for OKD and split out into separate PR: https://github.com/openshift/machine-config-operator/pull/3248

Requires: ~~https://github.com/openshift/os/pull/773 and possibly https://issues.redhat.com/browse/MCO-126~~ we can defer both

/cc @cheesesashimi 
/cc @travier 

**- How to verify it**

**- Description for the changelog**
MCD: Add support for RHCOS9 and SCOS9

### CODE REVIEW

File: pkg/daemon/osrelease.go
Comment (LorbusChris) [inline pkg/daemon/osrelease.go:38]: cc @travier

File: pkg/daemon/kernelargs.go
Comment (cheesesashimi) [inline pkg/daemon/kernelargs.go:31]: This is mostly for my own edification: Are these args no longer used?

**Comment by cheesesashimi** [comment]
@LorbusChris Is there any additional work required or could we merge this as-is? Overall, it seems reasonable to me modulo the discussions around the karg tuning allowlist.

**Comment by LorbusChris** [comment]
I think this is good to go. I'd like for @vrutkovs to chime in one the allowlist aspect.

**Comment by LorbusChris** [comment]
@cheesesashimi I've split the FCOS allowlist part out into https://github.com/openshift/machine-config-operator/pull/3248.
This PR here should be ready then.

**Review by travier — APPROVED** [review]

**Comment by LorbusChris** [comment]
/assign @sinnykumari
could you approve?

**Review by sinnykumari — APPROVED** [review]

File: pkg/daemon/update.go
Comment (travier) [inline pkg/daemon/update.go:1744]: I've just realized here that this is https://issues.redhat.com/browse/COS-1589 and this is not yet done.

**Review by travier — CHANGES_REQUESTED** [review]
See https://issues.redhat.com/browse/COS-1589

**Comment by LorbusChris** [comment]
Rebased and updated to include support for RHCOS 9

**Comment by LorbusChris** [comment]
Blocked on https://gitlab.com/redhat/centos-stream/rpms/ignition/-/merge_requests/11

File: pkg/daemon/update.go
Comment (LorbusChris) [inline pkg/daemon/update.go:1792]: NTS: fixup this comment - RHCOS9 is now accounted for in this PR

**Comment by bgilbert** [comment]
LGTM

**Review by travier — APPROVED** [review]

**Comment by LorbusChris** [comment]
@travier I'm seeing `e2e-agnostic-upgrade` failures on other PRs, too. I think it's unrelated. Do you think we could merge https://gitlab.com/redhat/centos-stream/rpms/ignition/-/merge_requests/11 today to unblock this PR?

**Comment by LorbusChris** [comment]
The ignition spec change has merged, and the package been rebuilt (https://kojihub.stream.centos.org/koji/buildinfo?buildID=24602) :tada: 

This should be good to go!

File: pkg/daemon/osrelease.go
Comment (cgwalters) [inline pkg/daemon/osrelease.go:23]: This is OK as is but...the better way to do this I think is to use the `ID_LIKE` field - it's intended for exactly this use case.  IOW we'd check whether `strings.Contains(os.ID_LIKE, "rhel")` (or to do it a bit more precisely, split `ID_LIKE` into a proper whitespace separated array of strings).

In particular, this would make things Just Work on non-Stream CentOS, Alma, etc.

Comment (LorbusChris) [reply pkg/daemon/osrelease.go:23]: hm, wouldn't that only be safe in combination with `VARIANT_ID == "coreos"`? (which RHCOS8 doesn't specify)

Comment (LorbusChris) [reply pkg/daemon/osrelease.go:23]: The func isn't particularly well named, it probably should've been `IsELCOS()`

**Pull request merged**
